imap_processing/ultra/l1c/make_helio_index_maps.py

```diff
@@ -0,0 +1,335 @@
+"""Make heliocentric HEALPix index maps for Ultra L1C processing."""
+
+import logging
+
+import astropy_healpix.healpy as hp
+import numpy as np
+import spiceypy as sp
+import xarray as xr
+
+from imap_processing.spice.geometry import (
+    SpiceBody,
+    SpiceFrame,
+    get_rotation_matrix,
+    imap_state,
+)
+from imap_processing.ultra.constants import UltraConstants
+from imap_processing.ultra.l1b.lookup_utils import is_inside_fov
+from imap_processing.ultra.l1c.l1c_lookup_utils import build_energy_bins
+
+logger = logging.getLogger(__name__)
+
+
+def vector_ijk_to_theta_phi(
+    inst_vecs: np.ndarray,
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Convert instrument vectors to theta/phi.
+
+    Parameters
+    ----------
+    inst_vecs : np.ndarray
+        Array of shape (n, 3) with components (x, y, z).
+
+    Returns
+    -------
+    theta : np.ndarray
+        Declination in radians, range [-π, π].
+    phi : np.ndarray
+        Right ascension in radians, range [-π, π].
+    """
+    # Extract components
+    i_comp = inst_vecs[:, 0]  # x component
+    j_comp = inst_vecs[:, 1]  # y component
+    k_comp = inst_vecs[:, 2]  # z component
+
+    # Normalize
+    magnitude = np.linalg.norm(inst_vecs, axis=1)
+
+    # Compute declination and right ascension
+    theta = np.arcsin(i_comp / magnitude)
+    phi = np.arctan2(j_comp, k_comp)
+
+    # Wrap to [-π, π]
+    theta = np.where(theta > np.pi, theta - 2 * np.pi, theta)
+    phi = np.where(phi > np.pi, phi - 2 * np.pi, phi)
+
+    return theta, phi
+
+
+def make_helio_index_maps_with_nominal_kernels(
+    kernel_paths: list[str],
+    nside: int,
+    spin_duration: float,
+    num_steps: int,
+    instrument_frame: SpiceFrame = SpiceFrame.IMAP_ULTRA_90,
+    compute_bsf: bool = False,
+    boundary_points: int = 8,
+) -> xr.Dataset:
+    """
+    Create index maps with nominal sim kernels.
+
+    This function ensures SPICE kernels are loaded before creating the maps. It uses
+    a KernelPool context manager to ensure only this function uses the nominal sim
+    kerneles.
+
+    Parameters
+    ----------
+    kernel_paths : list[str]
+        List of string paths to nominal simulated SPICE kernels.
+    nside : int
+        HEALPix nside parameter.
+    spin_duration : float
+        Total spin period in seconds.
+    num_steps : int
+        Number of spin phase steps.
+    instrument_frame : SpiceFrame, optional
+        Instrument frame (default IMAP_ULTRA_90).
+    compute_bsf : bool, optional
+        Compute boundary scale factors (default False).
+    boundary_points : int, optional
+        Number of boundary points per pixel (default 8).
+
+    Returns
+    -------
+    xr.Dataset
+        Dataset with helio index maps.
+    """
+    # Get all loaded SPK kernels
+    spk_kernels = [sp.kdata(i, "spk")[0] for i in range(sp.ktotal("spk"))]
+    # Find the de440s.bps kernel
+    de440s_file = next((k for k in spk_kernels if "de440s" in k), None)
```

Comment:
There are both the `de440.bsp` and `de440s.bsp` kernels. The `de440s.bsp` kernel has the "s" for shortened and excludes some far past and distant future coverage in order to make a smaller file. We use the "s" version in testing, but the full `de440.bsp` kernel in production. You should make this code find either one.
```suggestion
    de440s_file = next((k for k in spk_kernels if "de440" in k), None)
```